### ISSUE TITLE
Media flickering

### ISSUE THREAD

**Issue by Neurotoxin001** [telegramdesktop/tdesktop#17109]
### Steps to reproduce

1. Open any video/photo
2. Close this video/photo
3. Open any other video/photo
4. You will see for the 0.1 sec prevous video/photo

When I leafing through media with mousewheel or keyboard arrows there is no issue.

https://user-images.githubusercontent.com/39812401/137590280-85ff176b-f301-4d13-bbbf-0f68c9d16988.mp4

### Expected behaviour

There should be no flickering of previous media when opening new media

### Actual behaviour

There is flickering of previous media when opening new media

### Operating system

Windows 11 21H2 (22000.258)

### Version of Telegram Desktop

3.1.8

### Installation source

Static binary from official website

**Comment by Neurotoxin001** [comment]
Same on Windows 10 21H1

**Comment by Neurotoxin001** [comment]
Fixed when I set "angle graphics backend" from auto to disable.

![image](https://user-images.githubusercontent.com/39812401/138274161-b0a92ed2-cb3d-4499-ac13-e204968fe060.png)

**Comment by HarDX** [comment]
Same problem on Win 10 21H1
Tested "OpenGL", "Direct3D 9" - no issue
Tested "D3D11on12", "Direct3D 11" - have issue

**Comment by Aokromes** [comment]
> Same problem on Win 10 21H1 with video playback in full screen Tested "OpenGL", "Direct3D 9" - no issue Tested "D3D11on12", "Direct3D 11" - have issue

what's your video (gfx) card and what drivers version?

**Comment by HarDX** [comment]
> what's your video (gfx) card and what drivers version?
lenovo 330-15ikb - gpu intel 620 + nvidia mx150
Last Driver on both 
intel: 30.0.101.1069 (Date 11.02.2021)
nvidia: 30.0.14.9676 (496.76)
As i see i have problem today, maybe after last update telegram

**Comment by HarDX** [comment]
When not working
```
[2021.11.17 15:31:49] App Info: reading encrypted mtp config...
[2021.11.17 15:31:49] Export Info: Destroy top bar by controller removal.
[2021.11.17 15:31:49] OpenGL: Using custom 'gpu_driver_bug_list.json'.
[2021.11.17 15:31:51] OpenGL Profile: OpenGLES.
[2021.11.17 15:31:51] OpenGL Renderer: ANGLE (Intel, Intel(R) UHD Graphics 620 Direct3D11 vs_5_0 ps_5_0, D3D11-30.0.101.1069)
[2021.11.17 15:31:51] OpenGL Vendor: Google Inc. (Intel)
[2021.11.17 15:31:51] OpenGL Version: OpenGL ES 3.0.0 (ANGLE 2.1.7 git hash: 0bb011f9e403)
[2021.11.17 15:31:51] OpenGL Extensions: GL_NV_EGL_stream_consumer_external, , GL_ANGLE_texture_multisample, GL_ANGLE_framebuffer_multisample, GL_OES_fbo_render_mipmap, GL_OES_get_program_binary, GL_KHR_parallel_shader_compile, GL_OES_depth32, GL_ANGLE_instanced_arrays, GL_OES_compressed_ETC2_RGB8_texture, GL_OES_texture_half_float_linear, GL_ANGLE_texture_compression_dxt3, GL_NV_pack_subimage, GL_OES_texture_border_clamp, GL_ANGLE_copy_texture_3d, GL_EXT_texture_filter_anisotropic, GL_OES_compressed_ETC2_sRGB8_texture, GL_EXT_draw_buffers_indexed, GL_WEBGL_video_texture, GL_EXT_sRGB, GL_CHROMIUM_bind_uniform_location, GL_OES_compressed_EAC_R11_unsigned_texture, GL_EXT_texture_format_BGRA8888, GL_EXT_float_blend, GL_EXT_texture_compression_rgtc, GL_ANGLE_provoking_vertex, GL_OES_compressed_ETC2_RGBA8_texture, GL_OES_vertex_array_object, GL_ANGLE_depth_texture, GL_EXT_discard_framebuffer, GL_OES_texture_stencil8, GL_ANGLE_explicit_context_gles1, GL_EXT_unpack_subimage, GL_ANGLE_program_cache_control, GL_ANGLE_robust_client_memory, GL_CHROMIUM_sync_query, GL_EXT_texture_rg, GL_OES_texture_float, GL_OES_texture_half_float, GL_OVR_multiview, GL_ANGLE_get_tex_level_parameter, GL_OES_mapbuffer, GL_OES_compressed_ETC2_punchthroughA_sRGB8_alpha_texture, GL_OES_standard_derivatives, GL_KHR_debug, GL_ANGLE_request_extension, GL_OES_compressed_EAC_R11_signed_texture, GL_OES_draw_elements_base_vertex, GL_EXT_debug_label, GL_ANGLE_multiview_multisample, GL_EXT_instanced_arrays, GL_EXT_texture_compression_dxt1, GL_OES_compressed_ETC2_punchthroughA_RGBA8_texture, GL_EXT_map_buffer_range, GL_OES_surfaceless_context, GL_ANGLE_framebuffer_blit, GL_NV_fence, GL_EXT_disjoint_timer_query, GL_EXT_occlusion_query_boolean, GL_ANGLE_client_arrays, GL_EXT_draw_buffers, GL_EXT_debug_marker, GL_OVR_multiview2, GL_EXT_texture_compression_bptc, GL_EXT_robustness, GL_OES_compressed_ETC2_sRGB8_alpha8_texture, GL_ANGLE_translated_shader_source, GL_EXT_multisampled_render_to_texture, GL_ANGLE_lossy_etc_decode, GL_EXT_texture_norm16, GL_OES_texture_npot, GL_OES_compressed_EAC_RG11_signed_texture, GL_OES_EGL_image, GL_ANGLE_texture_usage, GL_EXT_read_format_bgra, GL_NV_pixel_buffer_object, GL_EXT_texture_storage, GL_ANGLE_memory_size, GL_EXT_color_buffer_float, GL_ANGLE_base_vertex_base_instance, GL_OES_EGL_image_external, GL_OES_EGL_image_external_essl3, GL_OES_element_index_uint, GL_CHROMIUM_copy_texture, GL_OES_compressed_EAC_RG11_unsigned_texture, GL_EXT_shader_texture_lod, GL_CHROMIUM_copy_compressed_texture, GL_OES_packed_depth_stencil, GL_OES_draw_buffers_indexed, GL_EXT_draw_elements_base_vertex, GL_ANGLE_texture_compression_dxt5, GL_CHROMIUM_lose_context, GL_OES_rgb8_rgba8, GL_ANGLE_explicit_context, GL_ANGLE_get_serialized_context_string, GL_ANGLE_multi_draw, GL_EXT_color_buffer_half_float, GL_CHROMIUM_bind_generates_resource, GL_EXT_frag_depth, GL_EXT_blend_func_extended, GL_EXT_blend_minmax, GL_ANGLE_pack_reverse_row_order, GL_OES_texture_float_linear, GL_EXT_texture_compression_s3tc_srgb, GL_EXT_EGL_image_external_wrap_modes, GL_OES_depth24
[2021.11.17 15:31:51] EGL Extensions: EGL_EXT_create_context_robustness, EGL_ANGLE_d3d_share_handle_client_buffer, EGL_ANGLE_d3d_texture_client_buffer, EGL_ANGLE_surface_d3d_texture_2d_share_handle, EGL_ANGLE_query_surface_pointer, EGL_ANGLE_window_fixed_size, EGL_ANGLE_keyed_mutex, EGL_ANGLE_surface_orientation, EGL_ANGLE_direct_composition, EGL_ANGLE_windows_ui_composition, EGL_NV_post_sub_buffer, EGL_KHR_create_context, EGL_KHR_image, EGL_KHR_image_base, EGL_KHR_gl_texture_2D_image, EGL_KHR_gl_texture_cubemap_image, EGL_KHR_gl_renderbuffer_image, EGL_KHR_get_all_proc_addresses, EGL_KHR_stream, EGL_KHR_stream_consumer_gltexture, EGL_NV_stream_consumer_gltexture_yuv, EGL_ANGLE_flexible_surface_compatibility, EGL_ANGLE_stream_producer_d3d_texture, EGL_ANGLE_create_context_webgl_compatibility, EGL_CHROMIUM_create_context_bind_generates_resource, EGL_CHROMIUM_sync_control, EGL_EXT_pixel_format_float, EGL_KHR_surfaceless_context, EGL_ANGLE_display_texture_share_group, EGL_ANGLE_display_semaphore_share_group, EGL_ANGLE_create_context_client_arrays, EGL_ANGLE_program_cache_control, EGL_ANGLE_robust_resource_initialization, EGL_ANGLE_create_context_extensions_enabled, EGL_ANDROID_blob_cache, EGL_ANDROID_recordable, EGL_ANGLE_image_d3d11_texture, EGL_ANGLE_create_context_backwards_compatible, EGL_KHR_create_context_no_error, EGL_KHR_reusable_sync, 
[2021.11.17 15:31:51] OpenGL: QOpenGLContext created, version: 3.3.
[2021.11.17 15:31:51] OpenGL: [TRUE] (OverlayWidget)
[2021.11.17 15:31:57] Audio Info: recreating audio device and reattaching the tracks
[2021.11.17 15:32:02] Audio Info: Closing audio playback device.
[2021.11.17 15:32:02] Audio Info: recreating audio device and reattaching the tracks
[2021.11.17 15:32:07] Audio Info: Closing audio playback device.
```

**Comment by HarDX** [comment]
When works on OpenGL normal
```
[2021.11.17 15:34:32] Map read time: 3
[2021.11.17 15:34:32] App Info: reading encrypted mtp config...
[2021.11.17 15:34:32] Export Info: Destroy top bar by controller removal.
[2021.11.17 15:34:32] OpenGL: Using custom 'gpu_driver_bug_list.json'.
[2021.11.17 15:34:34] OpenGL: Could not create widget in a window.
[2021.11.17 15:34:34] OpenGL: [FALSE] (OverlayWidget)
[2021.11.17 15:34:38] Audio Info: recreating audio device and reattaching the tracks
[2021.11.17 15:34:41] Audio Info: Closing audio playback device.
```

**Comment by HarDX** [comment]
When works on Dx9 normal
```
[2021.11.17 15:35:59] App Info: reading encrypted mtp config...
[2021.11.17 15:36:00] Export Info: Destroy top bar by controller removal.
[2021.11.17 15:36:00] OpenGL: Using custom 'gpu_driver_bug_list.json'.
[2021.11.17 15:36:00] OpenGL Profile: OpenGLES.
[2021.11.17 15:36:00] OpenGL Renderer: ANGLE (Intel, Intel(R) UHD Graphics 620 Direct3D9Ex vs_3_0 ps_3_0, igdumdim64.dll-30.0.101.1069)
[2021.11.17 15:36:00] OpenGL Vendor: Google Inc. (Intel)
[2021.11.17 15:36:00] OpenGL Version: OpenGL ES 2.0.0 (ANGLE 2.1.7 git hash: 0bb011f9e403)
[2021.11.17 15:36:00] OpenGL Extensions: GL_ANGLE_texture_usage, GL_CHROMIUM_bind_uniform_location, , GL_CHROMIUM_bind_generates_resource, GL_EXT_occlusion_query_boolean, GL_EXT_float_blend, GL_NV_read_depth, GL_EXT_read_format_bgra, GL_EXT_texture_filter_anisotropic, GL_EXT_texture_storage, GL_OES_texture_npot, GL_CHROMIUM_copy_texture, GL_ANGLE_get_tex_level_parameter, GL_WEBGL_video_texture, GL_ANGLE_depth_texture, GL_OES_EGL_image_external, GL_OES_vertex_array_object, GL_ANGLE_texture_compression_dxt5, GL_OES_standard_derivatives, GL_CHROMIUM_lose_context, GL_EXT_texture_compression_dxt1, GL_ANGLE_instanced_arrays, GL_NV_pack_subimage, GL_OES_texture_half_float, GL_ANGLE_request_extension, GL_OES_element_index_uint, GL_OES_depth32, GL_OES_texture_float, GL_ANGLE_framebuffer_blit, GL_CHROMIUM_sync_query, GL_EXT_debug_marker, GL_ANGLE_framebuffer_multisample, GL_ANGLE_memory_size, GL_OES_texture_stencil8, GL_EXT_frag_depth, GL_OES_packed_depth_stencil, GL_ANGLE_pack_reverse_row_order, GL_ANGLE_translated_shader_source, GL_ANGLE_robust_client_memory, GL_CHROMIUM_color_buffer_float_rgb, GL_EXT_texture_format_BGRA8888, GL_EXT_unpack_subimage, GL_EXT_shader_texture_lod, GL_ANGLE_explicit_context_gles1, GL_EXT_color_buffer_half_float, GL_OES_rgb8_rgba8, GL_OES_texture_float_linear, GL_ANGLE_program_cache_control, GL_NV_fence, GL_NV_read_stencil, GL_ANGLE_get_serialized_context_string, GL_EXT_blend_minmax, GL_ANGLE_client_arrays, GL_EXT_debug_label, GL_OES_surfaceless_context, GL_ANGLE_texture_compression_dxt3, GL_OES_get_program_binary, GL_ANGLE_multi_draw, GL_ANGLE_explicit_context, GL_EXT_robustness, GL_KHR_parallel_shader_compile, GL_CHROMIUM_color_buffer_float_rgba, GL_OES_texture_border_clamp, GL_KHR_debug, GL_OES_EGL_image, GL_OES_texture_half_float_linear
[2021.11.17 15:36:00] EGL Extensions: EGL_EXT_create_context_robustness, EGL_ANGLE_d3d_share_handle_client_buffer, EGL_ANGLE_d3d_texture_client_buffer, EGL_ANGLE_surface_d3d_texture_2d_share_handle, EGL_ANGLE_query_surface_pointer, EGL_ANGLE_window_fixed_size, EGL_NV_post_sub_buffer, EGL_KHR_create_context, EGL_KHR_image, EGL_KHR_image_base, EGL_KHR_gl_texture_2D_image, EGL_KHR_gl_renderbuffer_image, EGL_KHR_get_all_proc_addresses, EGL_ANGLE_flexible_surface_compatibility, EGL_ANGLE_create_context_webgl_compatibility, EGL_CHROMIUM_create_context_bind_generates_resource, EGL_EXT_pixel_format_float, EGL_KHR_surfaceless_context, EGL_ANGLE_display_texture_share_group, EGL_ANGLE_display_semaphore_share_group, EGL_ANGLE_create_context_client_arrays, EGL_ANGLE_program_cache_control, EGL_ANGLE_robust_resource_initialization, EGL_ANGLE_create_context_extensions_enabled, EGL_ANDROID_blob_cache, EGL_ANDROID_recordable, EGL_ANGLE_create_context_backwards_compatible, EGL_KHR_create_context_no_error, EGL_KHR_reusable_sync, 
[2021.11.17 15:36:00] OpenGL: QOpenGLContext created, version: 2.2.
[2021.11.17 15:36:00] OpenGL: [TRUE] (OverlayWidget)
[2021.11.17 15:36:05] Audio Info: recreating audio device and reattaching the tracks
[2021.11.17 15:36:08] Audio Info: Closing audio playback device.
```

**Comment by Neurotoxin001** [comment]
`+`

**Comment by Neurotoxin001** [comment]
Issue is still here

**Comment by Neurotoxin001** [comment]
Update.

• ANGLE Direct3D 9 with **enabled** HW accelerated video decoding: videos is **green;**
• ANGLE Direct3D 9 with **disabled** HW accelerated video decoding: videos is **pink;**

All other ANGLE's (D3D 11 / 11on12) **is fine** with enabled and with disabled HW accelerated video decoding.

ANGLE OpenGL = Disabled (no animation when media rotate).

And **no more media flickering,** so bug is fixed for me. Tested on 3.8.10.

**Comment by Neurotoxin001** [comment]
Turns out the problem is not fixed even in 4.8.11. But it started to appear randomly and not often.

**Comment by lankylonky22** [comment]
> Turns out the problem is not fixed even in 4.8.11. But it started to appear randomly and not often.

its still here, use d3d9 instead of auto solved the media flickering previous item issue for me

**Comment by lankylonky22** [comment]
is there a way to use opengl rendering on windows

**Comment by Neurotoxin001** [comment]
OpenGL was disabled because it's even not working (it's been same as "Disabled"):
https://github.com/telegramdesktop/tdesktop/commit/c5718678ea9ffb7e82c8c3d19e36b4a9080965f7

For me on 4.9.2 these options is fine:
![image](https://github.com/telegramdesktop/tdesktop/assets/39812401/305c8f81-c088-48d6-8478-0b5c27a4e741)

Flickering is **still not fixed,** but it started to appear randomly and **not often.**

**Comment by lankylonky22** [comment]
> OpenGL was disabled because it's even not working (it's been same as "Disabled"): [c571867](https://github.com/telegramdesktop/tdesktop/commit/c5718678ea9ffb7e82c8c3d19e36b4a9080965f7)
> 
> For me on 4.9.2 these options is fine: ![image](https://user-images.githubusercontent.com/39812401/264242063-305c8f81-c088-48d6-8478-0b5c27a4e741.png)
> 
> Flickering is **still not fixed,** but it started to appear randomly and **not often.**

nah this bug appears every time a pic is displayed

**Comment by Neurotoxin001** [comment]
Can you record video?

**Comment by alexeynl** [comment]
I have the same issue. 

Video demonstration:

https://www.youtube.com/watch?v=ZdN6GR_VYgM

Timings: 5, 21, 24, 27 seconds.

Current Telegram version installed 4.11.7. Windows 10 and 11.

1. I have this issue on my ryzen 6800h laptop since i have bought it in july 2022.
2. I tried a lot of things since i july 2022: starting from reinstall windows (including windows 10) and ends with regular AMD drivers, Windows and Telegram updates.
3. Issue is not reproduced with Angle disabled. See the same video demostration starting from 40 seconds.
4. There is no issue on Linux on the same laptop.
5. Laptop has not standard display resolution 3200x2000 (16:10)

I had opened issue before:
https://github.com/telegramdesktop/tdesktop/issues/27130
